src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -5,29 +5,21 @@
 using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
-using System.Composition;
 using System.Linq;
 using System.Runtime.CompilerServices;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp.Extensions;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Editing;
-using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.CodeAnalysis.LanguageServices;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Simplification;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CSharp.CodeGeneration
 {
-    [ExportLanguageService(typeof(SyntaxGenerator), LanguageNames.CSharp), Shared]
-    internal class CSharpSyntaxGenerator : SyntaxGenerator
+    internal partial class CSharpSyntaxGenerator : SyntaxGenerator
     {
-        [ImportingConstructor]
-        public CSharpSyntaxGenerator()
-        {
-        }
```

Comment:
note that this is moved into a separate part.  This allows hte data-only portion of this type (i.e. all of it by the mef stuff) to be used as a singleton in the codestyle layer.  The workspcae layer then languageservice-izes it so it can be imported from that point upwards.